[R-package] clarified error messages and documentation for lgb.get.eval.result()

jameslamb opened the pull request:
In this PR, I propose some more user-friendly documentation and error messages for `lgb.get.eval_result()`. I think it would be helpful to be more explicit about how someone can find valid values of `data_name` and `eval_name`.

I've also added tests on the validations in this function, since currently they are not covered by tests.

StrikerRUS commented on R-package/R/lgb.Booster.R:
```diff
@@ -930,13 +940,30 @@ lgb.get.eval.result <- function(booster, data_name, eval_name, iters = NULL, is_
+      "lgb.get.eval.result: data_name "
+      , shQuote(data_name)
+      , " not found. Only the followwing datasets exist in record evals: ["
+      , paste(data_names, collapse = ",")
```
```suggestion
      , paste(data_names, collapse = ", ")
```
I think it will be more readable.

jameslamb replied:
sure

StrikerRUS commented on R-package/R/lgb.Booster.R:
```diff
@@ -930,13 +940,30 @@ lgb.get.eval.result <- function(booster, data_name, eval_name, iters = NULL, is_
+      , " not found. Only the followwing eval_names exist for dataset "
+      , shQuote(data_name)
+      , ": ["
+      , paste(eval_names, collapse = ",")
```
```suggestion
      , paste(eval_names, collapse = ", ")
```

StrikerRUS commented on R-package/R/lgb.Booster.R:
```diff
@@ -930,13 +940,30 @@ lgb.get.eval.result <- function(booster, data_name, eval_name, iters = NULL, is_
+    stop(paste0(
+      "lgb.get.eval.result: data_name "
+      , shQuote(data_name)
+      , " not found. Only the followwing datasets exist in record evals: ["
```
```suggestion
      , " not found. Only the following datasets exist in record evals: ["
```

StrikerRUS commented on R-package/R/lgb.Booster.R:
```diff
@@ -930,13 +940,30 @@ lgb.get.eval.result <- function(booster, data_name, eval_name, iters = NULL, is_
+    stop(paste0(
+      "lgb.get.eval.result: eval_name "
+      , shQuote(eval_name)
+      , " not found. Only the followwing eval_names exist for dataset "
```
```suggestion
      , " not found. Only the following eval_names exist for dataset "
```

StrikerRUS commented on R-package/man/lgb.get.eval.result.Rd:
```diff
@@ -13,23 +13,26 @@ lgb.get.eval.result(
 }
 \arguments{
-\item{booster}{Object of class \code{lgb.Booster}}
+\item{booster}{Object of class \code{gb.Booster}}
```
```suggestion
\item{booster}{Object of class \code{lgb.Booster}}
```

StrikerRUS commented on R-package/tests/testthat/test_lgb.Booster.R:
```diff
@@ -0,0 +1,90 @@
+            , data_name = "testing"
+            , eval_name = "l2"
+        )
+    }, regexp = "Only the followwing datasets exist in record evals: [test]", fixed = TRUE)
```
```suggestion
    }, regexp = "Only the following datasets exist in record evals: [test]", fixed = TRUE)
```

StrikerRUS commented on R-package/tests/testthat/test_lgb.Booster.R:
```diff
@@ -0,0 +1,90 @@
+            , data_name = "test"
+            , eval_name = "l1"
+        )
+    }, regexp = "Only the followwing eval_names exist for dataset 'test': [l2]", fixed = TRUE)
```
```suggestion
    }, regexp = "Only the following eval_names exist for dataset 'test': [l2]", fixed = TRUE)
```

jameslamb commented:
> Seems that the latest force-push reverted all fixes applied after review.

ah sorry! Will fix right now

jameslamb commented:
> > Seems that the latest force-push reverted all fixes applied after review.
> 
> ah sorry! Will fix right now

ok @StrikerRUS I've added the changes back in https://github.com/microsoft/LightGBM/pull/2686/commits/0e0b1cdaa0db4d763fa93627ca008e4928501ba5 and rebased to current `master` since #2688 was merged

StrikerRUS commented:
@jameslamb Thanks!
